test/e2e/framework/pod/utils.go

```diff
@@ -37,14 +37,9 @@ func NodeOSDistroIs(distro string) bool {
 }
 
 // GenerateScriptCmd generates the corresponding command lines to execute a command.
-// Depending on the Node OS is Windows or linux, the command will use powershell or /bin/sh
 func GenerateScriptCmd(command string) []string {
 	var commands []string
-	if !NodeOSDistroIs("windows") {
```

Comment:
this function is used by some functions. Need to check and make adjustment accordingly, I think. 

eg, runVolumeTesterPod() function

Reply:
yes, my idea was to see what breaks in the Presubmit test, it looks like most if not all of the powershell commands also work with /bin/sh

Reply:
thanks for finding this one, I saw that it fails in agnhost: 
```
C:/ $ while(1) {sleep 2}
bash: syntax error: unexpected word (expecting "do")
```

Reply:
is there any other places you need to fix like runVolumeTesterPod()?

Reply:
I think I only checked usages of `GenerateScriptCmd` but some methods were calling it with a command sent from outside like `MakePod`

```
func MakePod(ns string, nodeSelector map[string]string, pvclaims []*v1.PersistentVolumeClaim, isPrivileged bool, command string) *v1.Pod {
	....
		Spec: v1.PodSpec{
			Containers: []v1.Container{
				{
					Name:            "write-pod",
					Image:           GetDefaultTestImage(),
					Command:         GenerateScriptCmd(command),
					SecurityContext: GenerateContainerSecurityContext(isPrivileged),
				},
			},
			RestartPolicy: v1.RestartPolicyOnFailure,
		},
	}
```

I'd need to do check usages of methods that also use `GenerateScriptCmd`, I'll do another scan today

Reply:
I've checked usages of `GenerateScriptCmd`:

- `MakePod` -> [`CreatePod`, `CreateUnschedulablePod`]
- `MakeSecPod`
- `makeStateuflSetWithPVCs`
- `StartInPodWithInlineVolume` 
- `StartInPodWithVolume` (fixed a few issues here)
- `setInitCommand` (I found a lot of powershell commands here but the test are marked as `[LinuxOnly]`, you might take a look in `subpath.go` @jingxu97), after more debugging in `subpath.go` I found more calls to powershell but the calls are not using `GenerateScriptCmd`, instead they have their own function to either call powershell or /bin/sh

After doing the above I analyzed each file inside `testsuites` looking for `NodeOSDistroIs("windows")`, there were none.

There's common code that's outside `testsuites/` like `storage/framework/` or `framework/volume` which I also checked, it's kinda hard to tell how many places are affected by this change until we check CI